source/core/index.ts

```diff
@@ -927,7 +929,9 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 			}
 		}
 
-		if (defaults && !areHooksDefault) {
+		const hooksNormalizedAlready = options && (options as NormalizedOptions)[kHooksNormalizedAlready];
```

Comment:
Can you use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining here?

Reply:
Ooh neat, I didn't know about that. Done! Also slightly tweaked the tests that I added.